Add Fir generation support for Heroku Private Spaces (#401)
heroku/terraform-provider-heroku: docs/resources/space.md

Provides a Heroku Private Space resource for running apps in isolated, highly available, secure app execution environments.

> Heroku Private Spaces support two generations:

Reviewer: ```suggestion
Both generations of the Heroku platform offer Private Spaces:
```

> * **Cedar** (default): The original generation supporting all Private Space features including Shield spaces

Reviewer: ```suggestion
* **Cedar** (default): The [Cedar generation](https://devcenter.heroku.com/articles/private-spaces#additional-features-for-cedar-private-spaces) supports all Private Space features, including Shield spaces.
* ```

* **Fir**: The next-generation platform with enhanced capabilities for Cloud Native Buildpacks (CNB), but with some feature limitations

> ~> **Note:** The `generation` parameter cannot be changed after space creation. Choose carefully based on your application requirements.

Reviewer: ```suggestion
~> **Note:** You can't change the `generation` parameter after space creation. Choose carefully based on your application requirements.
```

## Example Usage

A Heroku "team" was originally called an "organization", and that is still 
the identifier used in this resource.

```hcl-terraform

> // Create a new Cedar generation space (default)

Reviewer: ```suggestion
// Create a new Cedar-generation space (default)
```

resource "heroku_space" "cedar_space" {
  name = "test-cedar-space"
  organization = "my-company"
  region = "virginia"

>   shield = true  // Cedar supports shield spaces

Reviewer: ```suggestion
  shield = true  // Cedar supports Shield spaces
```

}

> // Create a new Fir generation space

Reviewer: ```suggestion
// Create a new Fir-generation space
```

* `id` - The ID (UUID) of the space.
* `name` - The space's name.
* `organization` - The space's Heroku Team.

> * `generation` - The space's generation (`cedar` or `fir`).

Reviewer: ```suggestion
* `generation` - The generation of the Heroku platform for the space (`cedar` or `fir`).
```
